dm/loader/util.go

```diff
@@ -141,15 +143,52 @@ func getMydumpMetadata(cli *clientv3.Client, cfg *config.SubTaskConfig, workerNa
 	return loc.Position.String(), loc.GTIDSetStr(), nil
 }
 
+// getMydumpMetadataByExternalStorage gets Metadata by ExternalStorage.
+func getMydumpMetadataByExternalStorage(ctx context.Context, cli *clientv3.Client, cfg *config.SubTaskConfig, workerName string) (string, string, error) {
+	metafile := "metadata"
+	loc, _, err := dumpling.ParseMetaDataByExternalStore(ctx, cfg.Dir, metafile, cfg.Flavor)
+	if err != nil {
```

Comment:
we can check `if err == nil { return }` to reduce one indent level.

And if `os.IsNotExist`, it may reach `toPrint, err2 := exstorage.ReadFile(ctx, cfg.Dir, metafile)`, but will `exstorage.ReadFile` always fail because the file is not exists?

Reply:
ok, let me test about this.

Reply:
After some test about this, it will not renturn `os.IsNotExist` when call s3 api, it will return like https://github.com/pingcap/tidb/blob/master/br/pkg/storage/s3.go#L422, and i use `exstorage.IsNotExistError` to check this error.

Reply:
I prefer

```
if err == nil {
  return loc.Position.String(), loc.GTIDSetStr(), nil
}

if storage.IsNotExistError(err) {
  // getLoadTask to check if a DM-worker scheduling is needed.
  // if not found a scheduling, return with some file-not-found message, no need to go to rest logic
}
if terror.ErrMetadataNoBinlogLoc.Equal(err) {
  // log and return
}
// get the file content and return
```

Reply:
ok, that will be more concise.